taqueria-vscode-extension/src/test/suite/extension.test.ts

```diff
@@ -5,102 +5,112 @@ import * as vscode from 'vscode';
+	mockedReturnValues.openDialogMockValues = mockedPaths;
+	mockedReturnValues.openDialogMockIndex = 0;
+}
+
+function getNextOpenDialogUri() {
+	if (
+		!mockedReturnValues.openDialogMockValues
+		|| mockedReturnValues.openDialogMockIndex >= mockedReturnValues.openDialogMockValues.length
+	) {
+		throw new Error('We have already used all mocked values.');
+	}
+	const mockedValue = mockedReturnValues.openDialogMockValues[mockedReturnValues.openDialogMockIndex];
+	mockedReturnValues.openDialogMockIndex++;
+	return mockedValue;
+}
+
+let choosePlugin: string = '@taqueria/plugin-smartpy';
+
+const mockedMethods = {
+	'window.showInformationMessage': (msg: string) => Promise.resolve(console.log(msg)),
+	'window.showQuickPick': () => Promise.resolve([choosePlugin]),
+	'window.showOpenDialog': () => {
+		return Promise.resolve([Uri.file(getNextOpenDialogUri())]);
+	},
+};
+vscodeMock = MockedObject.make(vscode, mockedMethods);
+
+describe('Extension Test Suite', async () => {
 	before(async () => {
 		const context: ExtensionContext = {
 			subscriptions: [],
 		} as any;
-		await taqueriaExtension.activate(context, {
-			vscode: MockedObject.make(vscode, {
-				'window.showInformationMessage': (msg: string) => Promise.resolve(console.log(msg)),
-			}),
-		});
-		await fse.copy(testProjectSource, testProjectDestination);
+
+		await taqueriaExtension.activate(context, { vscode: vscodeMock });
 
 		vscode.window.showInformationMessage('Start all tests.');
 	});
 
-	it('Verify that Taqueria Initiate will init new taquifed  project ', async () => {
+	it('Verify that Taqueria commands present in the command pallet list ', async () => {
 		await vscode.commands.getCommands(true).then(allCommands => {
 			const taqCommands = allCommands.filter(command => command.toLowerCase().includes('taq'));
 			assert.notEqual(taqCommands, undefined);
 		});
-
-		// TODO: Need to figure out if it is possible to interact with showDialogBox
-		//      If no this test cannot be automated
-		//      I tried to use RobotJS, but it has an issue to work with parser properly
-		//      Even if I exclude it it keeps asking to rebuild it due to an old version of npm
-		//      https://stackoverflow.com/questions/46384591/node-was-compiled-against-a-different-node-js-version-using-node-module-versio
-
-		// await vscode.commands.executeCommand("taqueria.init").then( ()=>{
-		// });
-
-		// Example how to use sleep to stop VS Code for a moment
-		// await sleep(60000);
 	});
 
-	// TODO: https://github.com/ecadlabs/taqueria/issues/605
-	it.skip('Verify that VS Code command Taqueria Scaffold will scaffold a project', async () => {
-		// Open taquified project in workspace under testProjectDestination
+	it('Verify that VS Code command Taqueria Compile Ligo will compile Ligo contract', async () => {
+		// It creates another process
+		// https://stackoverflow.com/questions/51385812/is-there-a-way-to-open-a-workspace-from-an-extension-in-vs-code
```

Comment:
I think a lot of this commended out code can be removed since it doesn't seem to be relevant

Reply:
@alexzbusko I would keep it because it is a reminder of why it cannot be done this way. I spent a lot of time debugging it, and it is nice to have a reminder about it.